src/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -0,0 +1,195 @@
+				int end = offset + part.length();
+				if (LOGGER.isDebugEnabled()) {
+					LOGGER.debug("Name: " + name + " O: " + offset + " S: " + start + " E: " + end + " Pos: " + position);
+				}
+				multiPartParams.add(new MultipartFormParameter(name, valueMatcher.group("value"), start, end, position,
+						MultipartFormParameter.Type.GENERAL));
+				if (LOGGER.isDebugEnabled()) {
+					LOGGER.debug("Group name: " + name + " value: " + valueMatcher.group("value"));
+				}
+				if (partHeaderLine.contains("filename=")) {
+					// Extract the filename
+					Matcher fnValueMatcher = FILENAME_PART_PATTERN.matcher(part);
+					fnValueMatcher.find();
+					String fnValue = fnValueMatcher.group("filename");
+					extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, fnValue, ++position));
+					int fnStart = offset + part.indexOf(fnValue);
+					int fnEnd = offset + (fnStart + fnValue.length());
+					if (LOGGER.isDebugEnabled()) {
+						LOGGER.debug("Name: " + name + " O: " + offset + " S: " + fnStart + " E: " + fnEnd + " Pos: " + position);
+					}
+					multiPartParams.add(new MultipartFormParameter(name, fnValue, fnStart, fnEnd, position,
+							MultipartFormParameter.Type.FILE_NAME));
+					// Extract the content-type
+					Matcher ctValueMatcher = CONTENTTYPE_PART_PATTERN.matcher(part);
+					ctValueMatcher.find();
+					String ctValue = ctValueMatcher.group("contenttype");
+					extractedParameters.add(new NameValuePair(NameValuePair.TYPE_POST_DATA, name, ctValue, ++position));
+					int ctStart = offset + part.indexOf(ctValue);
+					int ctEnd = offset + (ctStart + ctValue.length());
+					if (LOGGER.isDebugEnabled()) {
+						LOGGER.debug("Name: " + name + " O: " + offset + " S: " + ctStart + " E: " + ctEnd + " Pos: " + position);
+					}
+					multiPartParams.add(new MultipartFormParameter(name, ctValue, ctStart, ctEnd, position,
+							MultipartFormParameter.Type.FILE_CONTENT_TYPE));
+				}
+			}
+			position++;
+			offset = offset + part.length();
+		}
+		params = Collections.unmodifiableList(extractedParameters);
+	}
+
+	@Override
+	public List<NameValuePair> getParamList() {
+		return params;
+	}
+
+	@Override
+	public String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+		return setParameter(msg, originalPair, name, value, false);
+	}
+
+	@Override
+	public String setEscapedParameter(HttpMessage msg, NameValuePair originalPair, String name, String value) {
+		return setParameter(msg, originalPair, name, value, true);
+	}
+
+	private String setParameter(HttpMessage msg, NameValuePair originalPair, String name, String value,
+			boolean escaped) {
+		String contentType = msg.getRequestHeader().getHeader(HttpHeader.CONTENT_TYPE);
+		String boundary = getBoundary(contentType);
+
+		StringBuilder newBodyBuilder = new StringBuilder(msg.getRequestBody().toString());
+		for (int idx = 0; idx < params.size(); idx++) {

```

Comment:
The loop can be removed now (and just replace).

Reply:
The bus ness is being used to get the param from the list, I guess I could change it to a map using param name as the key?

Reply:
Right, but the list has the same order (less one) as the parameters so we can just access it (`int idx = originalPair.getPosition() -1; MultipartFormParameter mpPart = multiPartParams.get(idx);`).

Reply:
Sorry been working with the same code too long to see an alternative (some sort of “functional fixed ness”). Thanks @thc202 !